### PR TITLE
Improve Openapi v3 compliance 

### PR DESCRIPTION
**Pull request by RawToast**
## Summary by CodeRabbit

* **New Features**
  * Enhanced discriminated unions and discriminator mappings
  * Improved additionalProperties handling with typed catchalls, nested support, and value constraints
  * Better support for complex schema composition (allOf/anyOf/oneOf), nested defaults, recursion, and not/refinements
  * Richer non-JSON response handling (binary, streams, varied content types)
  * Array bounds, uniqueness and refined constraints applied
  * Property metadata support (readOnly, writeOnly, const, default, deprecated, examples)

* **Tests**
  * Extensive new test suites validating all above schema-generation scenarios and edge cases

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: packages/zenko/src/__tests__/oneof-discriminator.test.ts [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+      },
+    } as OpenAPISpec
+    const result = generate(specYaml)
+
+    expect(result).toContain('z.literal("foo_kind")')
+    expect(result).toContain('z.literal("foo_alias")')
+    expect(result).toContain('z.literal("extra_kind")')
+    expect((result.match(/Foo\.merge/g) ?? []).length).toBe(3)
+    expect(result).toContain("Extra.merge")
+  })
```
[the remark this answers is not in the document]
Comment (RawToast) [reply packages/zenko/src/__tests__/oneof-discriminator.test.ts]: https://github.com/RawToast/zenko/pull/53/commits/07d8e79b7750039a9ab8b33314518eac03b3a9a2

[the remark this answers is not in the document]
Comment (RawToast) [reply packages/zenko/src/__tests__/oneof-discriminator.test.ts]: https://github.com/RawToast/zenko/pull/53/commits/7c897066c8efaaa2ea2654084357133404c729e8
